packages/ipfs/src/http/utils/joi.js

```diff
@@ -130,20 +165,25 @@ module.exports = Joi
             throw new Error('Invalid base name')
           }
 
-          return value
+          return { value }
         }
       }
     },
     (joi) => {
       return {
-        name: 'json',
-        base: joi.string(),
-        pre (value, state, options) {
+        type: 'json',
+        base: joi.any(),
+        validate (value, helpers) {
+          if (helpers.schema.$_getFlag('presence') === 'required' && !value) {
```

Comment:
Should we simplify this validation with a function? It seems replicated several times

Reply:
Probably. TBH I'm not sure why this is necessary, extending joi is really poorly documented.

Reply:
yeah, totally